Algorithmia/CLI.py

```diff
@@ -0,0 +1,233 @@
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=algo_input,
+					headers={'Authorization':key.read(),'Content-Type':'application/json'}, params= algo.query_parameters).json())
+				key.close()
+			
+			elif(inputs[0] == "-b"):
+				#binary
+				algo_input = inputs[1]
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=bytes(algo_input),
+					headers={'Authorization':key.read(),'Content-Type':'application/octet-stream'}, params= algo.query_parameters).json())
+				key.close()
+			
+			elif(inputs[0] == "-D" or inputs[0] == "--data-file"):
+				#data file
+				algo_input = open(inputs[1],"r").read()
+				try:
+					result = algo.pipe(algo_input)
+				except Exception as error:
+					print(error)
+
+			elif(inputs[0] == "-T"):
+				#text file
+				algo_input = open(inputs[1],"r").read()
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+				
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=algo_input.encode('utf-8'),
+					headers={'Authorization':key.read(),'Content-Type':'text/plain'}, params= algo.query_parameters).json())
+				key.close()
+			
+			elif(inputs[0] == "-J"):
+				#json file
+				#read json file and run algo with that input bypassing the auto detection of input type in pipe
+				algo_input = open(inputs[1],"r").read()
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=json.dumps(algo_input).encode('utf-8'),
+					headers={'Authorization':key.read(),'Content-Type':'application/json'}, params= algo.query_parameters))
+				key.close()
+
+			elif(inputs[0] == "-B"):
+				#binary file
+				algo_input = open(inputs[1],"rb").read()
+				print(algo_input)
+				print("end input \n")
+				key = open(os.environ['HOME']+"/.algorithmia_api_key","r")
+
+				result = AlgoResponse.create_algo_response(requests.post(client.apiAddress + algo.url, data=bytes(algo_input),
+					headers={'Authorization':key.read(),'Content-Type':'application/octet-stream'}, params= algo.query_parameters).json())
+				key.close()
+
+			else:
+				algo_input = inputs[0]
+				try:
+					result = algo.pipe(algo_input)
+				except Exception as error:
+					print(error)
+			
+			#handle output flags
+
+			#response
+
+		#response-body
+		if(inputs[-1] == "--response-body"):
+			result_body = """{ "result": """+result.result + ",\n" + """  "metadata": """ 
+			result_body = result_body + json.dumps(result.metadata.full_metadata)
+			result = result_body + " }"
+
+		#output to file if there is an output file spesified
+		elif(len(inputs) >= 3):
+			if(inputs[-2] == "--output" or inputs[-2] == "-o"):
+				try:
+					if isinstance(result.result, bytearray) or isinstance(result.result, bytes):
+						out = open(inputs[-1],"wb")
```

Comment:
When using files in python, best practice is to use `with open()`. That will handle open files and exceptions better.